src/scanner.c

```diff
@@ -195,6 +195,10 @@ void Match (
 {
     Char                errmsg [256];
 
+    if (STATE(InterpreterStartLine) == 0) {
+        STATE(InterpreterStartLine) = STATE(SymbolStartLine);
```

Comment:
BTW, `InterpreterStartLine` might be incorrect if there is a recursive interpreter call (right at its start, resp. right after it ended). So perhaps `InterpreterStartLine` should be stored and restored along with Symbol and other interpreter state. Unfortunately, that whole business of storing interpreter state is something I have not yet had time to clean up and refactor (though it is on my implicit TODO); as a result, `Symbol` is (re)stored via `IO()->Input->symbol` in `io.c`; while `StackObj` is (re)stored via `PushPlist(STATE(IntrState), STATE(StackObj));` in `intrprtr.c`; and other bits and pieces are (re)stored via various code in `read.c`.

Anyway, that's hypothetical, I don't know whether this really manifests as a problem.

Reply:
I am saving/restoring in io.c, in the same place as Symbol is. I started thinking about trying to neaten this up, then got scared away, congratulations on your progress in cleaning up this area, it's very hairy :)